extension/src/openvic-extension/singletons/MenuSingleton.cpp

```diff
@@ -452,6 +502,157 @@ int32_t MenuSingleton::get_rgo_owner_pop_icon_index() const {
 	return sprite;
 }
 
+/* TOPBAR */
+
+Dictionary MenuSingleton::get_topbar_info() const {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V(game_singleton, {});
+
+	CountryInstance const* country = game_singleton->get_viewed_country();
+	if (country == nullptr) {
+		return {};
+	}
+
+	Dictionary ret;
```

Comment:
Why dictionary instead of class/struct?

Reply:
A custom struct would have to be registered with Godot to be usable at the GDScript level. In cases like this that's a lot more code to write for little gain (GDScript looks up class member variables by name in the same way as Dictionary entries are looked up, so if there's any performance difference I'd expect classes to be the slower approach).